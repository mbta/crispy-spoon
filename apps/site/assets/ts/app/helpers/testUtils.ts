/* eslint-disable */
import toJson, { Json } from "enzyme-to-json";
import { ReactWrapper } from "enzyme";
import {
  IndexedRouteStop,
  StopTree
} from "../../schedule/components/__schedule";

export const testRouteStopListFromStopTree = (
  tree: StopTree
): IndexedRouteStop[] =>
  Object.values(tree.byId).map((node, index) => ({
    ...node.value,
    routeIndex: index
  }));

export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};

export const enzymeToJsonWithoutProps = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false,
    map: info => {
      if (!info.props) {
        return info;
      }

      // @ts-ignore is a <style></style> declaration
      if (info.node.type === "style") {
        return info;
      }
      // @ts-ignore is an element in a <noscript></noscript> declaration
      if (info.node.nodeType == undefined) {
        // @ts-ignore
        return info;
      }
      //@ts-ignore is a React component (not HTML element)
      if (info.node.nodeType.toString().includes("function")) {
        // @ts-ignore
        info.props = undefined;
      }

      return info;
    }
  });

export const toJsonEnzyme = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false
  });
