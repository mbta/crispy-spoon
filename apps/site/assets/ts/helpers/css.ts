import {
  isABusRoute,
  isACommuterRailRoute,
  isAGreenLineRoute,
  isASilverLineRoute
} from "../models/route";
import { Route } from "../__v3api";

export const routeToCSSClass = (route: string): string =>
  route.toLowerCase().replace(/[ _]/g, "-");

export const busClass = (route: Route): string =>
  isABusRoute(route) && !isASilverLineRoute(route) ? "bus-route-sign" : "";

export const routeToModeName = (route: Route): string => {
  if (isACommuterRailRoute(route)) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red" || route.id === "Mattapan") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (isAGreenLineRoute(route)) return "green-line";
  if (isASilverLineRoute(route)) return "silver-line";
  if (isABusRoute(route)) return "bus";
  return "unknown";
};

export const routeBgClass = (route: Route): string =>
  `u-bg--${routeToModeName(route)}`;
