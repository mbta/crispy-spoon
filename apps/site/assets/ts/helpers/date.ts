// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a date in the form of "Aug 7"
export const shortDate = (date: Date): string =>
  date.toLocaleDateString("en-US", {
    month: "short",
    day: "numeric",
    timeZone: "UTC"
  });

export default formattedDate;
