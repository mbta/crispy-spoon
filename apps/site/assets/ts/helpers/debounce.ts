// Debounce is meant to be used with event listeners. Use it to delay a call
// until after a burst of events, such as keyboard input.
//
//   usage:
//
//     const callback = (ev: Event) => {}
//     document.addEventListener("keyup", debounce(callback, 250))
//

const debounce = (callback: Function, delay: number): EventListener => {
  let timer: number | undefined;

  return (...args) => {
    window.clearTimeout(timer);
    timer = window.setTimeout(() => callback(...args), delay);
  };
};

export const debouncePromise = (
  callback: Function,
  delay: number
): Function => {
  let timerId: number | undefined;

  return function debounced(...args: unknown[]) {
    if (timerId) {
      window.clearTimeout(timerId);
    }

    return new Promise(resolve => {
      timerId = window.setTimeout(() => resolve(callback(...args)), delay);
    });
  };
};

export default debounce;
