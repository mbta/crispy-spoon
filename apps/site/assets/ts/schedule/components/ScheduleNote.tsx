import React, { ReactElement } from "react";
import {
  ScheduleNote as ScheduleNoteType,
  ServiceException
} from "./__schedule";
import scheduleIcon from "../../../static/images/icon-schedule-finder.svg";

const service = (serviceTime: string): ReactElement<HTMLElement> => (
  <p className="m-schedule-page__service-note-time" key={serviceTime}>
    Trains arrive every {serviceTime}
  </p>
);

const offPeak = (
  offpeak: string,
  exceptions: ServiceException[]
): ReactElement<HTMLElement> => {
  const except =
    exceptions.length > 0
      ? `except ${exceptions.map(e => e.type).join(", ")}`
      : "";
  return service(`${offpeak} ${except}`);
};

interface Props {
  scheduleNote: ScheduleNoteType;
}

const ScheduleNote = ({
  scheduleNote: {
    peak_service: peakService,
    offpeak_service: offpeakService,
    exceptions
  }
}: Props): ReactElement<HTMLElement> => (
  <div className="m-schedule-page__schedule-note-container">
    <h2 className="m-schedule-page__schedule-note-title">
      <div
        className="m-schedule-page__schedule-note-icon"
        dangerouslySetInnerHTML={{ __html: scheduleIcon }}
      />
      Schedule Note
    </h2>
    <div className="m-schedule-page__schedule-note">
      <h4 className="m-schedule-page__service">Peak Service</h4>
      <div className="m-schedule-page__service-subheading">
        Weekdays 6:30AM - 9AM, 3:30PM - 6:30 PM
      </div>
      {service(peakService)}
    </div>
    <div className="m-schedule-page__schedule-note">
      <h4 className="m-schedule-page__service">Off Peak / Weekends</h4>
      {offPeak(offpeakService, exceptions)}
      {exceptions.map(exception =>
        service(`${exception.service} ${exception.type}`)
      )}
    </div>
  </div>
);

export default ScheduleNote;
