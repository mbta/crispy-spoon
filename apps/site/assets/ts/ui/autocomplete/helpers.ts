import { omit } from "lodash";
import { OnSubmitParams } from "@algolia/autocomplete-core";
import { OnStateChangeProps } from "@algolia/autocomplete-js";
import {
  ContentItem,
  Item,
  LocationItem,
  PopularItem,
  RouteItem,
  StopItem
} from "./__autocomplete";
import { isLGDown } from "../../helpers/media-breakpoints";

export function isStopItem(x: Item): x is StopItem {
  return Object.keys(x).includes("stop");
}

export function isRouteItem(x: Item): x is RouteItem {
  return Object.keys(x).includes("route");
}

export function isContentItem(x: Item): x is ContentItem {
  return Object.keys(x).includes("_content_type");
}

export const getTitleAttribute = (item: Item): string[] => {
  if (isStopItem(item)) {
    return ["stop", "name"];
  }
  if (isRouteItem(item)) {
    return ["route", "name"];
  }

  // @ts-ignore
  // eslint-disable-next-line no-underscore-dangle
  return Object.keys(item._highlightResult);
};

const navStateChange: (props: OnStateChangeProps<Item>) => void = ({
  state
}) => {
  // grey out the page and disable scrolling when search is open
  if (isLGDown()) {
    if (state.isOpen) {
      document.documentElement.dataset.navOpen = "true";
    } else {
      delete document.documentElement.dataset.navOpen;
    }
  }
};

export const STATE_CHANGE_HANDLERS: Record<
  string,
  (props: OnStateChangeProps<Item>) => void
> = {
  nav: navStateChange
};

export const SUBMIT_HANDLERS: Record<
  string,
  (props: OnSubmitParams<Item>) => void
> = {
  to_search_page({ state }) {
    window.Turbolinks.visit(`/search?query=${state.query}`);
  }
};

export type WithUrls<T> = T & { urls: Record<string, string> };

// The backend returns all possible URLs, use urlType to get the desired one
export const itemWithUrl = (
  initialItem: WithUrls<LocationItem | PopularItem>,
  urlType: string
): Omit<typeof initialItem, "urls"> => {
  const item = omit(initialItem, "urls");
  item.url = initialItem.urls[urlType];
  return item;
};
