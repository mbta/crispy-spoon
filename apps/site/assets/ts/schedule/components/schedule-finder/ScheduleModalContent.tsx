import React, { ReactElement } from "react";
import { DirectionId, Route } from "../../../__v3api";
import { routeToModeName } from "../../../helpers/css";
import {
  SimpleStopMap,
  RoutePatternsByDirection,
  ServiceInSelector,
  SelectedOrigin,
  UserInput,
  ScheduleNote
} from "../__schedule";
import { EnhancedJourney, Journey, TripInfo } from "../__trips";
import ScheduleFinderForm from "./ScheduleFinderForm";
import DailySchedule from "./daily-schedule/DailySchedule";
import UpcomingDepartures from "./upcoming-departures/UpcomingDepartures";
import { useProvider } from "../../../helpers/use-provider";
import {
  fetchJsonOrThrow,
  fetchJson,
  isFetchFailed
} from "../../../helpers/fetch-json";
import { useAwaitInterval } from "../../../helpers/use-await-interval";
import { isSubwayRoute } from "../../../models/route";
import DailyScheduleSubway from "./daily-schedule/DailyScheduleSubway";
import formattedDate, { stringToDateObject } from "../../../helpers/date";
import { isInCurrentService } from "../../../helpers/service";

// exported for testing
export const fetchData = async (
  routeId: string,
  selectedOrigin: string,
  selectedDirection: DirectionId,
  date: string
): Promise<EnhancedJourney[]> => {
  const departures = await fetchJsonOrThrow<Journey[]>(
    `/schedules/finder_api/departures?id=${routeId}&stop=${selectedOrigin}&direction=${selectedDirection}`
  );

  const enhanced = await Promise.all(
    departures.map(async departure => {
      const res = await fetchJson<TripInfo>(
        `/schedules/finder_api/trip?id=${departure.trip.id}&route=${routeId}&date=${date}&direction=${selectedDirection}&stop=${selectedOrigin}`
      );

      if (isFetchFailed(res)) {
        // 404s here are a known failure mode, see finder_api.ex#get_trip_info
        if (res.status !== 404) {
          throw new Error(
            `Failed to fetch trip information: ${res.status} ${res.statusText}`
          );
        }

        return { ...departure, tripInfo: null };
      }

      return { ...departure, tripInfo: res };
    })
  );

  return enhanced;
};

interface Props {
  handleChangeDirection: (direction: DirectionId) => void;
  handleChangeOrigin: (origin: SelectedOrigin) => void;
  handleOriginSelectClick: () => void;
  route: Route;
  selectedDirection: DirectionId;
  selectedOrigin: string;
  services: ServiceInSelector[];
  stops: SimpleStopMap;
  routePatternsByDirection: RoutePatternsByDirection;
  today: string;
  scheduleNote: ScheduleNote | null;
}

const ScheduleModalContent = ({
  handleChangeDirection,
  handleChangeOrigin,
  handleOriginSelectClick,
  route,
  selectedDirection,
  selectedOrigin,
  services,
  stops,
  routePatternsByDirection,
  today,
  scheduleNote
}: Props): ReactElement<HTMLElement> | null => {
  const { id: routeId } = route;

  const input: UserInput = {
    route: routeId,
    origin: selectedOrigin,
    date: today,
    direction: selectedDirection
  };

  const [state, updateData] = useProvider(fetchData, [
    routeId,
    selectedOrigin,
    selectedDirection,
    input.date
  ]);
  useAwaitInterval(updateData, 10000);

  const serviceToday = services.some(service =>
    isInCurrentService(service, stringToDateObject(today))
  );

  const renderUpcomingDepartures = (): ReactElement<HTMLElement> =>
    serviceToday ? (
      <UpcomingDepartures state={state} />
    ) : (
      <div className="callout text-center u-bold">
        There are no scheduled trips for {formattedDate(today)}.
      </div>
    );

  return (
    <>
      <div className="schedule-finder schedule-finder--modal">
        <ScheduleFinderForm
          onDirectionChange={handleChangeDirection}
          onOriginChange={handleChangeOrigin}
          onOriginSelectClick={handleOriginSelectClick}
          route={route}
          selectedDirection={selectedDirection}
          selectedOrigin={selectedOrigin}
          stopsByDirection={stops}
        />
      </div>

      {!isSubwayRoute(route) ? null : (
        <DailyScheduleSubway
          stops={stops}
          services={services}
          stopId={selectedOrigin}
          directionId={selectedDirection}
          routeId={routeId}
          route={route}
          scheduleNote={scheduleNote}
          today={today}
        />
      )}

      {routeToModeName(route) !== "ferry" && renderUpcomingDepartures()}

      {isSubwayRoute(route) ? null : (
        <DailySchedule
          selectedOrigin={selectedOrigin}
          services={services}
          routeId={routeId}
          directionId={selectedDirection}
          routePatterns={routePatternsByDirection[selectedDirection]}
          today={today}
        />
      )}
    </>
  );
};

export default ScheduleModalContent;
