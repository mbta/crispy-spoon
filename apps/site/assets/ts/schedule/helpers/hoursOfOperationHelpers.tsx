import React, { ReactElement } from "react";
import renderFa from "../../helpers/render-fa";
import { SchedulePDF } from "../components/__schedule";

const pdfLink = (pdf: SchedulePDF | null): ReactElement<HTMLElement> | null => {
  if (!pdf) {
    return null;
  }
  return (
    <div className="fs-16 pt-16">
      <a
        key={pdf.url}
        href={pdf.url}
        rel="noopener noreferrer"
        className="c-call-to-action"
        target="_blank"
      >
        {`Open full schedule and map PDF`}{" "}
        {renderFa("fa-arrow-up-right-from-square", "")}
      </a>
    </div>
  );
};

export default pdfLink;
