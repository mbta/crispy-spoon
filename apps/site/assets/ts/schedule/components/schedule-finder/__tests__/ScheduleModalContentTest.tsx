import React from "react";
import renderer, { act } from "react-test-renderer";
import { EnhancedRoute } from "../../../../__v3api";
import ScheduleModalContent, { fetchData } from "../ScheduleModalContent";
import { SimpleStop, SimpleStopMap } from "../../__schedule";
import { EnhancedJourney } from "../../__trips";
import departuresResponse from "./test-data/departures.json";
import ScheduleNote from "../../ScheduleNote";
import { createReactRoot } from "../../../../app/helpers/testUtils";

const today = "2019-12-05";
const route: EnhancedRoute = {
  alerts: [],
  description: "",
  direction_destinations: { 0: "Oak Grove", 1: "Forest Hills" },
  direction_names: { 0: "Inbound", 1: "Outbound" },
  header: "",
  id: "Orange",
  name: "Orange",
  long_name: "Orange Line",
  type: 1
};

const greenRoute: EnhancedRoute = {
  alerts: [],
  description: "",
  direction_destinations: { 0: "East", 1: "West" },
  direction_names: { 0: "East", 1: "West" },
  header: "",
  id: "Green",
  name: "Green",
  long_name: "Green Line",
  type: 0
};

const oneDirectionRoute: EnhancedRoute = {
  alerts: [],
  description: "",
  direction_destinations: { 0: "Oak Grove", 1: null },
  direction_names: { 0: "Inbound", 1: null },
  header: "",
  id: "Orange",
  name: "Orange",
  long_name: "Orange Line",
  type: 1
};

const scheduleNoteData = {
  offpeak_service: "8-12 minutes",
  peak_service: "5 minutes",
  exceptions: [
    { service: "26 minutes", type: "weekend mornings and late night" }
  ],
  alternate_text: null
};

const stopList: SimpleStop[] = [
  { name: "Malden Center", id: "place-mlmnl", is_closed: false, zone: "1" },
  { name: "Wellington", id: "place-welln", is_closed: false, zone: "2" }
];

const stops: SimpleStopMap = { 0: stopList, 1: stopList.slice().reverse() };

const payload: EnhancedJourney[] = (departuresResponse as unknown) as EnhancedJourney[];

describe("ScheduleModalContent", () => {
  it("renders", () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ScheduleModalContent
          handleChangeDirection={() => {}}
          handleChangeOrigin={() => {}}
          handleOriginSelectClick={() => {}}
          route={route}
          stops={stops}
          selectedOrigin={stopList[0].id}
          selectedDirection={0}
          services={[]}
          routePatternsByDirection={{}}
          today={today}
          scheduleNote={null}
        />
      );
    });

    expect(tree).toMatchSnapshot();
  });

  it("renders with schedule note if present", () => {
    createReactRoot();
    const tree = renderer.create(
      <ScheduleModalContent
        handleChangeDirection={() => {}}
        handleChangeOrigin={() => {}}
        handleOriginSelectClick={() => {}}
        route={route}
        stops={stops}
        selectedOrigin={stopList[0].id}
        selectedDirection={0}
        services={[]}
        routePatternsByDirection={{}}
        today={today}
        scheduleNote={scheduleNoteData}
      />
    );
    expect(
      tree.root.findByType(ScheduleNote).props.scheduleNote.offpeak_service
    ).toBe("8-12 minutes");
  });

  describe("fetchData", () => {
    it("fetches data", () => {
      const spy = jest.fn();
      window.fetch = jest.fn().mockImplementation(
        () =>
          new Promise((resolve: Function) =>
            resolve({
              json: () => payload,
              ok: true,
              status: 200,
              statusText: "OK"
            })
          )
      );

      return fetchData("1", "99", 0, spy).then(() => {
        expect(window.fetch).toHaveBeenCalledWith(
          "/schedules/finder_api/departures?id=1&stop=99&direction=0"
        );
        expect(spy).toHaveBeenCalledWith({
          type: "FETCH_STARTED"
        });
        expect(spy).toHaveBeenCalledWith({
          type: "FETCH_COMPLETE",
          payload
        });
      });
    });

    it("fails gracefully if fetch is unsuccessful", () => {
      const spy = jest.fn();
      window.fetch = jest.fn().mockImplementation(
        () =>
          new Promise((resolve: Function) =>
            resolve({
              json: () => "Internal Server Error",
              ok: false,
              status: 500,
              statusText: "INTERNAL SERVER ERROR"
            })
          )
      );

      return fetchData(route.id, stopList[0].id, 0, spy).then(() => {
        expect(window.fetch).toHaveBeenCalledWith(
          "/schedules/finder_api/departures?id=Orange&stop=place-mlmnl&direction=0"
        );
        expect(spy).toHaveBeenCalledWith({
          type: "FETCH_STARTED"
        });
        expect(spy).toHaveBeenCalledWith({
          type: "FETCH_ERROR"
        });
      });
    });
  });
});
