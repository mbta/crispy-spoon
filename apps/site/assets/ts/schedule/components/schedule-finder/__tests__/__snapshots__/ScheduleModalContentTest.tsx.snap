// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleModalContent renders 1`] = `
Array [
  <div
    className="schedule-finder schedule-finder--modal"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="schedule-finder__heading"
      >
        <span
          aria-hidden="true"
          className="notranslate c-svg__icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
         Schedule Finder
      </h2>
      <div
        className="schedule-finder__prompt"
      >
        Choose a stop to get schedule information and real-time departure predictions.
      </div>
      <div
        className="schedule-finder__inputs"
      >
        <label
          className="schedule-finder__label"
        >
          Choose a direction
          <div
            className="c-select-custom__container "
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={0}
          >
            <select
              className="c-select-custom notranslate"
              data-testid="schedule-finder-direction-select"
              onChange={[Function]}
              value={0}
            >
              <option
                aria-label="INBOUND 
  Oak Grove"
                value={0}
              >
                INBOUND 
  Oak Grove
              </option>
              <option
                aria-label="OUTBOUND 
  Forest Hills"
                value={1}
              >
                OUTBOUND 
  Forest Hills
              </option>
            </select>
            <span
              aria-hidden="true"
              className="notranslate c-svg__icon c-select-custom__arrow"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </div>
        </label>
        <label
          className="schedule-finder__label"
        >
          Choose an origin stop
          <div
            className="c-select-custom__container "
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={0}
          >
            <select
              className="c-select-custom c-select-custom--noclick notranslate"
              onChange={[Function]}
              value="place-mlmnl"
            >
              <option
                value=""
              >
                Select
              </option>
              <option
                aria-label="Malden Center"
                value="place-mlmnl"
              >
                Malden Center
              </option>
              <option
                aria-label="Wellington"
                value="place-welln"
              >
                Wellington
              </option>
            </select>
            <span
              aria-hidden="true"
              className="notranslate c-svg__icon c-select-custom__arrow"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </div>
        </label>
      </div>
      <div
        className="schedule-finder__submit text-right"
      >
        <input
          className="btn btn-primary"
          type="submit"
          value="Get schedules"
        />
      </div>
    </form>
  </div>,
  <div>
    <div
      className="u-highlight-gray m-n24"
    >
      <div
        className="m-24"
      >
        <div
          className="d-flex pt-10"
        >
          <span
            aria-hidden="false"
            className="notranslate c-svg__icon schedule__icon-header--size me-8"
            dangerouslySetInnerHTML={
              Object {
                "__html": "SVG",
              }
            }
          />
          <div
            className="fs-18 u-bold"
          >
            Malden Center
          </div>
        </div>
        <div
          className="fs-12 u-bold pb-10"
        >
          To 
          Oak Grove
        </div>
      </div>
    </div>
    <h3
      className="pt-18"
    >
      Daily Schedule
    </h3>
    <div
      className="pt-8"
    >
      <div
        className="c-select-custom__container "
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <select
          className="c-select-custom notranslate"
          onChange={[Function]}
          value="weekday"
        >
          <option
            value="weekday"
          >
            Weekday 
            (Today)
          </option>
          <option
            value="saturday"
          >
            Saturday 
            
          </option>
          <option
            value="sunday"
          >
            Sunday 
            
          </option>
        </select>
        <span
          aria-hidden="true"
          className="notranslate c-svg__icon c-select-custom__arrow"
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
      </div>
    </div>
    <div
      aria-live="polite"
      className="d-flex justify-content-space-between pt-8"
      style={
        Object {
          "gap": "2%",
        }
      }
    >
      <div
        className="w-100 u-highlight ps-16 pt-16 pb-16"
      >
        <div
          className="fs-14"
        >
          First Train
        </div>
        <div
          className="fs-18 u-bold"
        />
      </div>
      <div
        className="w-100 u-highlight ps-16 pt-16 pb-16"
      >
        <div
          className="fs-14"
        >
          Last Train
        </div>
        <div
          className="fs-18 u-bold"
        />
      </div>
    </div>
    <div
      className="d-flex pt-8 pb-18 fs-18"
    >
      <a
        className="btn btn-secondary btn-block mt-8"
        href="/trip-planner/from/"
      >
        Plan Your Trip
      </a>
    </div>
  </div>,
  <div
    className="callout text-center u-bold"
  >
    There are no scheduled trips for 
    December 5, 2019
    .
  </div>,
]
`;
